src/modules/home/routes/home.route.tsx

```diff
@@ -9,14 +9,15 @@ import { useSearchParamsState } from '../hooks/use-search-params-state';
 
 export function HomeRoute() {
   const [selectedCategory, setSelectedCategory] = useSearchParamsState('category', '');
+  const [searchQuery, setsearchQuery] = useSearchParamsState('search', '');
 
   return (
     <Box sx={homeRouteStyles.container}>
-      <HeaderComponent />
+      <HeaderComponent setSelectedLocation={setsearchQuery} />
```

Comment:
You are not searching by location, so I think the prop should have a different name